esp-hal/src/clock/mod.rs

```diff
@@ -666,3 +670,76 @@ impl<'d> RadioClockController<'d> {
         clocks_ll::reset_rpa();
     }
 }
+
+#[doc(hidden)]
+/// These are all modems with bitmasks as their representation.
+pub enum ModemBitmask {
+    #[cfg(wifi)]
+    WiFi = 1,
+    #[cfg(bt)]
+    Bt = 2,
+    #[cfg(ieee802154)]
+    Ieee802154 = 4
+}
+
+/// Tracks which modems currently request an active PHY clock.
+///
+/// Using an AtomicU8 wrapped in a critical section mutex, isntead of a RefCell, prevents panic
+/// machinery from being generated.
+static PHY_CLOCK_REFS: critical_section::Mutex<AtomicU8> = critical_section::Mutex::new(AtomicU8::new(0));
```

Comment:
You don't need the critical section around an atomic u8

Reply:
Looking at it now, that makes sense.